File: rb/features.rb
# Weather SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module WeatherFeatures
  def self.make_feature(name)
    case name
    when "base"
      WeatherBaseFeature.new
    when "test"
      WeatherTestFeature.new
    else
      WeatherBaseFeature.new
    end
  end
end

File: py/features.py
# Weather SDK feature factory

from feature.base_feature import WeatherBaseFeature
from feature.test_feature import WeatherTestFeature


def _make_feature(name):
    features = {
        "base": lambda: WeatherBaseFeature(),
        "test": lambda: WeatherTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
